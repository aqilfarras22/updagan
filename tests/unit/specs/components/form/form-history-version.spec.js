import BootstrapVue from 'bootstrap-vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';

import formHistoryVersionComponent from '@/components/form/form-history-version';

const { it, describe, expect } = global;

const testPropsData = {
  formVersion: {
    primaryReason: 'primaryReason',
    sections: [{ id: 1 }],
  },
};

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('components/form/form-history-version.vue', () => {
  const createMountedWrapper = (options = {}) => shallowMount(formHistoryVersionComponent, {
    localVue,
    propsData: { ...testPropsData },
    stubs: {
      FormFieldsList: {
        name: 'FormFieldsList',
        render: h => h('b', 'FormFieldsList'),
      },
    },
    ...options,
  });
  it('should create instance', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('should have described child components', () => {
    expect(formHistoryVersionComponent.components.FormFieldsList).toBeTruthy();
  });

  it('should able to receive props', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.fakeProp).toBeUndefined();
    expect(wrapper.vm.formVersion).toEqual(testPropsData.formVersion);
  });

  it('should have method which emit "back" event by click on link: back', () => {
    const wrapper = createMountedWrapper();
    // test link
    const link = wrapper.find('.backButton');
    expect(link.is('a')).toBe(true);

    link.trigger('click');
    // assert event has been emitted
    expect(wrapper.emitted().back).toBeTruthy();
    // assert event count
    expect(wrapper.emitted().back.length).toBe(1);
  });

  it('has the expected html structure', () => {
    const wrapper = createMountedWrapper();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
