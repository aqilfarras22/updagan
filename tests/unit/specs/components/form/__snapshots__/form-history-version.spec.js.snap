// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/form/form-history-version.vue has the expected html structure 1`] = `
<div>
  <b-row-stub
    class="border-bottom header-row"
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <a
        class="backButton"
        href=""
      >
        <i
          class="fa fa-caret-left fa-lg"
        />
          Back to history
      </a>
    </b-col-stub>
     
    <b-col-stub
      class="text-right"
      tag="div"
    >
      <i
        class="square"
      />
        
      <span
        class="info"
      >
        Edited fields
      </span>
    </b-col-stub>
  </b-row-stub>
   
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      class="mt-3"
      tag="div"
    >
      <p>
        <strong>
          Reason for change: primaryReason
        </strong>
      </p>
    </b-col-stub>
  </b-row-stub>
   
  <b
    donotopenquery=""
    sections="[object Object]"
    showchanged=""
  >
    FormFieldsList
  </b>
</div>
`;
